### PR TITLE
add test for rpc protocol path preix

### PR DESCRIPTION
**Pull request by AllanZhengYP**
*Description of changes:*

Followup to https://github.com/awslabs/smithy-typescript/pull/406. Add protocol test for RPC protocols path prefix support.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/awsJson1_0/endpoint-paths.smithy [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        protocol: awsJson1_0,
+        method: "POST",
+        uri: "/custom/",
+        body: "{}",
+        host: "example.com/custom",
+        appliesTo: "client"
+    }
+])
+
+operation HostWithPathOperation {}
```
Comment (adamthom-amzn) [inline smithy-aws-protocol-tests/model/awsJson1_0/endpoint-paths.smithy:24]: missing newline at end of file

File: smithy-aws-protocol-tests/model/awsQuery/endpoint-paths.smithy [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                Custom endpoints supplied by users can have paths""",
+        protocol: awsQuery,
+        method: "POST",
+        uri: "/custom/",
+        body: "Action=HostWithPathOperation&Version=2020-01-08",
+        host: "example.com/custom",
+        appliesTo: "client"
+    }
+])
+operation HostWithPathOperation {}
```
Comment (adamthom-amzn) [inline smithy-aws-protocol-tests/model/awsQuery/endpoint-paths.smithy:23]: Also missing a newline here

File: smithy-aws-protocol-tests/model/ec2Query/endpoint-paths.smithy [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                Custom endpoints supplied by users can have paths""",
+        protocol: ec2Query,
+        method: "POST",
+        uri: "/custom/",
+        body: "Action=HostWithPathOperation&Version=2020-01-08",
+        host: "example.com/custom",
+        appliesTo: "client"
+    }
+])
+operation HostWithPathOperation {}
```
Comment (adamthom-amzn) [inline smithy-aws-protocol-tests/model/ec2Query/endpoint-paths.smithy:23]: Also missing a newline at the end here

**Review by adamthom-amzn — CHANGES_REQUESTED** [review]
The commit message has a typo - "preix" should be "prefix"

**Review by JordonPhillips — APPROVED** [review]

**Review by adamthom-amzn — APPROVED** [review]

**Pull request merged**